File: swadharma/apps/web/src/types/patch_types.py
with open('assessment.ts', 'r') as f:
    content = f.read()

# Fix duplicate Herb interface and duplicate AssessmentState fields

content = content.replace("""export interface Herb {
  id: string;
  sanskritName: string;
  commonName: string;
  botanicalName: string;
  rasa: string[];
  virya: string;
  vipaka: string;
  prabhava: string;
  doshaMatrix: {
    vata: string;
    pitta: string;
    kapha: string;
  };
  organSystems: string[];
  useCases: string[];
  description: string;
}""", "")

content = content.replace("""export interface Herb {
  id: string;
  sanskritName: string;
  commonName: string;
  botanicalName: string;
  rasa: string[];
  virya: string;
  vipaka: string;
  prabhava?: string;
  doshaMatrix: DoshaMatrix;
  organSystems: string[];
  useCases: string[];
  description: string;
}""", "")

content = content.replace("""export interface Herb {
  id: string;
  sanskritName: string;
  commonName: string;
  botanicalName: string;
  rasa: string[];
  virya: string;
  vipaka: string;
  prabhava: string;
  doshaMatrix: Record<string, string>;
  organSystems: string[];
  useCases?: string[];
  description?: string;
  traditionalUses?: string[];
  cautions?: string[];
}""", """export interface Herb {
  id: string;
  sanskritName: string;
  commonName: string;
  botanicalName: string;
  rasa: string[];
  virya: string;
  vipaka: string;
  prabhava: string;
  doshaMatrix: Record<string, string>;
  organSystems: string[];
  useCases?: string[];
  description?: string;
  traditionalUses?: string[];
  cautions?: string[];
}""")

content = content.replace("""  herbSearchQuery: string;
  herbDoshaFilter: string;
  herbOrganFilter: string;
  setHerbSearchQuery: (query: string) => void;
  setHerbDoshaFilter: (dosha: string) => void;
  setHerbOrganFilter: (organ: string) => void;""", "")

with open('assessment.ts', 'w') as f:
    f.write(content)

File: patch_types.ts
import fs from 'fs';

const file = 'swadharma/apps/web/src/types/assessment.ts';
let content = fs.readFileSync(file, 'utf-8');

if (!content.includes('export interface Herb')) {
content += `

export interface Herb {
  id: string;
  sanskritName: string;
  commonName: string;
  botanicalName: string;
  category?: string[];
  rasa: string[];
  virya: string;
  vipaka: string;
  prabhava: string;
  doshaMatrix: DoshaMatrix;
  organSystems: string[];
  benefits?: string[];
  contraindications?: string[];
  dosage?: string;
  useCases?: string[];
  description?: string;
}
`;
}

if (!content.includes('export interface RoutineData')) {
content += `

export interface RoutineData {
  routines: Record<string, DoshaRoutine>;
  seasons: Record<string, SeasonAdjustment>;
}
`;
}

fs.writeFileSync(file, content);
